module/zfs/range_tree.c

```diff
@@ -251,9 +251,19 @@ range_tree_destroy(range_tree_t *rt)
 void
 range_tree_adjust_fill(range_tree_t *rt, range_seg_t *rs, int64_t delta)
 {
-	ASSERT3U(rs_get_fill(rs, rt) + delta, !=, 0);
-	ASSERT3U(rs_get_fill(rs, rt) + delta, <=, rs_get_end(rs, rt) -
-	    rs_get_start(rs, rt));
+	if (delta < 0 && delta * -1 >= rs_get_fill(rs, rt)) {
+		zfs_panic_recover("zfs: attempting to decrease fill below 0; "
+		    "probable double remove in segment [%llx:%llx]",
```

Comment:
I think that this also triggers if the fill goes to 0, which the old asserts also did.  So I think that the panic message should be adjusted to reflect that it might be decreasing *to* zero, not *below* zero.